patacrep/files.py

```diff
@@ -12,24 +12,25 @@
 
 LOGGER = logging.getLogger(__name__)
 
-def recursive_find(root_directory, extensions):
+def recursive_find(root_directory, extensions=None):
     """Recursively find files with the given extensions, from a root_directory.
 
     Return a list of files matching those conditions.
 
     Arguments:
-    - `extensions`: list of accepted extensions.
+    - `extensions`: list of accepted extensions (None means every file).
     - `root_directory`: root directory of the search.
     """
     if not os.path.isdir(root_directory):
         return []
 
     matches = []
-    pattern = re.compile(r'.*\.({})$'.format('|'.join(extensions)))
+    if extensions:
+        pattern = re.compile(r'.*\.({})$'.format('|'.join(extensions)))
```

Comment:
Ou encore : 
~~~python
if extension is None:
    pattern = re.compile('.*')
else:
    pattern = re.compile(r'.*\.({})$'.format('|'.join(extensions)))
~~~

Comme ça, un simple `if pattern.match()` englobe les deux cas.